db/src/tx.rs

```diff
@@ -749,28 +757,115 @@ fn conclude_tx(tx: Tx, report: TxReport) -> Result<(TxReport, PartitionMap, Opti
 ///
 /// This approach is explained in https://github.com/mozilla/mentat/wiki/Transacting.
 // TODO: move this to the transactor layer.
-pub fn transact<'conn, 'a, I>(conn: &'conn rusqlite::Connection,
+pub fn transact<'conn, 'a, 'id, I>(conn: &'conn rusqlite::Connection,
                               partition_map: PartitionMap,
                               schema_for_mutation: &'a Schema,
                               schema: &'a Schema,
-                              entities: I) -> Result<(TxReport, PartitionMap, Option<Schema>)>
+                              entities: I) -> Result<(TxReport, PartitionMap, Option<Schema>, AttributeSet)>
     where I: IntoIterator<Item=Entity> {
 
     let mut tx = start_tx(conn, partition_map, schema_for_mutation, schema)?;
-    let report = tx.transact_entities(entities)?;
-    conclude_tx(tx, report)
+    let (report, changes) = tx.transact_entities(entities)?;
+    conclude_tx(tx, report, changes)
 }
 
 /// Just like `transact`, but accepts lower-level inputs to allow bypassing the parser interface.
-pub fn transact_terms<'conn, 'a, I>(conn: &'conn rusqlite::Connection,
+pub fn transact_terms<'conn, 'a, 'id, I>(conn: &'conn rusqlite::Connection,
                                     partition_map: PartitionMap,
                                     schema_for_mutation: &'a Schema,
                                     schema: &'a Schema,
                                     terms: I,
-                                    tempid_set: InternSet<TempId>) -> Result<(TxReport, PartitionMap, Option<Schema>)>
+                                    tempid_set: InternSet<TempId>) -> Result<(TxReport, PartitionMap, Option<Schema>, AttributeSet)>
     where I: IntoIterator<Item=TermWithTempIds> {
-
     let mut tx = start_tx(conn, partition_map, schema_for_mutation, schema)?;
-    let report = tx.transact_simple_terms(terms, tempid_set)?;
-    conclude_tx(tx, report)
+    let (report, changes) = tx.transact_simple_terms(terms, tempid_set)?;
+    conclude_tx(tx, report, changes)
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct BatchedTransaction {
```

Comment:
Perhaps call this `TransactedReports`, and keep a mapping from txid to `TxReport`?